### PR TITLE
chore(auth): Migrate some types to AWS E/S/D

### PR DESCRIPTION
**Pull request by dnys1**
Migrates some Auth types to the AWS Equatable, Serializable, Debuggable stack.

---

**Stack**:
- #1724
- #1723
- #1722
- #1721
- #1720
- #1719

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Comment by dnys1** [comment]
I just realized this includes migration of the Dart example apps as well which is why it's so big. A mishap in rebasing..

File: packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart [hunk 1]
```diff
@@ -66,7 +66,6 @@ class CognitoUserAttributeKey extends UserAttributeKey
   List<Object?> get props => [
         // Cognito will lowercase these in API calls
         key.toLowerCase(),
-        readOnly,
```
Comment (haverchuck) [inline packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart:69]: Why are we removing this?

**Review by haverchuck — APPROVED** [review]
Left one question, otherwise LGTM

File: packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart (hunk 1 above)
Comment (dnys1) [reply packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart:69]: You'd never have two attributes with the same key but different readOnly markers so it doesn't provide any additional info beyond just the key. Further, AWSDebuggable uses these properties to render `toString` and it doesn't add any value there.

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**